### PR TITLE
Change the executor core calculation to take into account the cluster manager

### PR DESCRIPTION
**Pull request by tgravescs**
fixes https://github.com/NVIDIA/spark-rapids/issues/9271

The bug above was that our core calculation was wrongly return the number of cores on the host when the user ran spark on yarn without specifying the spark.executor.cores.  The default value on yarn when spark executor cores isn't specified is 1.

This fixes the executor core calculation to deal with different cluster manager defaults.  YARN and k8s both default to 1 core if the user doesn't set spark executor cores.  Standalone mode  defaults to use all the cores on the executor.  local mode has the special local[num_cores] and local_cluster mode takes [num workers, num cores, memory] (local-cluster[2,4,4096].

This special cases each of those. If the master isn't recognized it takes spark.executor.cores or defaults to 1.  I don't know how or when that would ever happen though as master should be defined.  The other option would be to have this throw an exception but figure that could be worse if it does happen as there would be no workaround.  I'm assuming we don't support mesos, if someone tried it, it would fall into the default case where it would use spark.executor.cores or 1 by default which seems like a fine default.

I manually tested this with following options:
local mode: local, local[1], local[4], local[*]
localcluster: local-cluster[2,4,6000]
yarn, k8s on eks, Databricks standalone: no spark.executor.cores specified and with it specified

All the above manual tests resulted in correct calculation of cores.

### CODE REVIEW

**Comment by tgravescs** [comment]
build

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala [hunk 1]
```diff
@@ -110,10 +111,39 @@ object RapidsPluginUtils extends Logging {
...
+          case m if m.startsWith("spark") =>
+            // STANDALONE
+            executorCoreConfOption.map(_.toInt).getOrElse(Runtime.getRuntime.availableProcessors)
+          case m if m.startsWith("local-cluster") =>
+            TrampolineUtil.getCoresInLocalMode(m, conf)
+          case m if m.startsWith("local") =>
+            TrampolineUtil.getCoresInLocalMode(m, conf)
+          case _ =>
+            val coresToUse = executorCoreConfOption.map(_.toInt).getOrElse(1)
+            logWarning(s"Master: ${masterOption.getOrElse("None")} is unknown, number of " +
```
Comment (jlowe) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala:135]:
```suggestion
            logWarning(s"Master: $m is unknown, number of " +
```

**Review by jlowe — APPROVED** [review]
Minor comment, lgtm.

**Comment by tgravescs** [comment]
build

**Review by jlowe — APPROVED** [review]

**Pull request merged**